### ISSUE TITLE
getMethod() doesn't work

### ISSUE THREAD

**Issue by roukmoute** [Roave/BetterReflection#270]
Hi!

I'm working on `dev-master` version and I can't make work your example.

Here is my simple code in `web/test.php`:

```php
$loader = require __DIR__ . '/../vendor/autoload.php';

class MyClass
{
    public function foo()
    {
        return 5;
    }
}

$classInfo = \Roave\BetterReflection\Reflection\ReflectionClass::createFromName('MyClass');
$classInfo->getMethod('foo')->setBodyFromClosure(function () {
    return 4;
});

$c = new MyClass();
var_dump($c->foo());
die();
```

I shows me 5 instead of 4.
Am I missing something?

Thanks

**Comment by asgrim** [comment]
Hi @roukmoute yep, looks like you are missing registering the class with the autoloader. Take a look at the demo script here:

https://github.com/Roave/BetterReflection/blob/master/demo/monkey-patching/index.php#L14

**Comment by roukmoute** [comment]
Thanks for your rapidity !

Do you help one last time for this code:

```php
require __DIR__ . '/../vendor/autoload.php';

use Roave\BetterReflection\Reflection\ReflectionClass;
use Roave\BetterReflection\Util\Autoload\ClassLoader;
use Roave\BetterReflection\Util\Autoload\ClassLoaderMethod\FileCacheLoader;

$loader = new ClassLoader(FileCacheLoader::defaultFileCacheLoader(__DIR__));

// Create the reflection first (without loading)
$classInfo = ReflectionClass::createFromName('MyClass');
$loader->addClass($classInfo);

// Override the body...!
$classInfo->getMethod('foo')->setBodyFromClosure(
    function () {
        return 4;
    }
);


class MyClass
{
    public function foo()
    {
        return 5;
    }
}

$c = new MyClass();
echo $c->foo() . "\n"; // should be 4...!?!??

die();
```

I've got this:
`Fatal error: Uncaught Roave\BetterReflection\Util\Autoload\Exception\ClassAlreadyLoaded: Class MyClass has already been loaded into memory so cannot be modified in /www-data/vendor/roave/better-reflection/src/Util/Autoload/Exception/ClassAlreadyLoaded.php:12`

**Comment by Ocramius** [comment]
The exception is quite self-explanatory: you need your class to be
autoloaded (and not yet have been loaded) in order for this to work.

**Comment by roukmoute** [comment]
Thanks @Ocramius.

But this is really strange because I have `Class MyClass has already been loaded`.
And you explain me `not yet have been loaded`.

This is not really the same thing.

**Comment by Ocramius** [comment]
BetterReflection intercepts the autoloading process in order to modify the
class on the fly: if autoloading already happened, there is nothing we can
do, and the class definition is already in the PHP engine.

On 24 Apr 2017 18:43, "Mathias STRASSER" <notifications@github.com> wrote:

Closed #270 <https://github.com/Roave/BetterReflection/issues/270>.

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/Roave/BetterReflection/issues/270#event-1055138891>,
or mute
the thread

**Comment by roukmoute** [comment]
Yep I've read that [here](https://github.com/Roave/BetterReflection/blob/master/docs/reflection-modification.md) :)

Thank you for taking the time to answer me.
